CheckPointFirewall Integration small updates

morita39 opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=12724) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=12724) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: morita39<br/>:white_check_mark: dantavori<br/>:x: cincrement<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=12724) it.</sub>

sakiyanagi commented:
Customer adds on reqeust, as the followings includes
    - writing IPv6 info into CheckPoint.Group.members context
    - add option of ignore warning/error for add-host command ( this enable to enter duplicated object)
    - suppress error from checkpoint_list_package_command, in case of return values does not match the JSON key defined
    - add 'no_more_output' message when checkpoint-host-list commands return output total:0

Noy-Maimon commented:
@morita39 Thank you for your contribution!
Please see my comments on places in the code that requires minor modifications.
Besides that, here are some general comments:
1. Please merge from `master` branch the changes made since forking.
2. Please add release notes for this version of the integration. you can do that via running `demisto-sdk update-release-notes -i Packs/CheckpointFirewall -u revision`. This will create an .md file where you should add a short description of the changes made, then commit and push.
3. Note that the made changes are failing some unit tests - please fix the unit tests to work with the current version of the integration.

morita39 commented:
> @morita39 Thank you for your contribution!
> Please see my comments on places in the code that requires minor modifications.
> Besides that, here are some general comments:
> 
> 1. Please merge from `master` branch the changes made since forking.
> 2. Please add release notes for this version of the integration. you can do that via running `demisto-sdk update-release-notes -i Packs/CheckpointFirewall -u revision`. This will create an .md file where you should add a short description of the changes made, then commit and push.
> 3. Note that the made changes are failing some unit tests - please fix the unit tests to work with the current version of the integration.
> 
> Feel free to contact me if something is unclear or if I can assist in anyway, thanks.

1. I sync up the forked repository with the master, the did pull request. I hope this is okay
2. I run `demisto-sdk update-release-notes -i Packs/CheckpointFirewall -u revision` , but error occurred. Not sure what to do.
`$ demisto-sdk update-release-notes -i Packs/CheckpointFirewall/  -u revision
You are using demisto-sdk 1.3.7.
Starting to update release notes.

General: [BA107] - Running on master branch while using git is ill advised.
run: 'git checkout -b NEW_BRANCH_NAME' and rerun the command.

Traceback (most recent call last):
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/git/util.py", line 889, in __getitem__
    return getattr(self, index)
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/git/util.py", line 882, in __getattr__
    return list.__getattribute__(self, attr)
AttributeError: 'IterableList' object has no attribute 'origin/master'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/morita/.pyenv/versions/3.9.0/bin/demisto-sdk", line 33, in <module>
    sys.exit(load_entry_point('demisto-sdk==1.3.7', 'console_scripts', 'demisto-sdk')())
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/demisto_sdk/__main__.py", line 1044, in update_release_notes
    modified, added, changed_meta_files, old = validate_manager.get_changed_files_from_git()
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/demisto_sdk/commands/validate/validate_manager.py", line 976, in get_changed_files_from_git
    modified_files = self.git_util.modified_files(prev_ver=self.prev_ver,
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/demisto_sdk/commands/common/git_util.py", line 45, in modified_files
    renamed = {item[0] for item in self.renamed_files(prev_ver, committed_only, staged_only)}
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/demisto_sdk/commands/common/git_util.py", line 283, in renamed_files
    deleted = self.deleted_files(prev_ver, committed_only, staged_only)
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/demisto_sdk/commands/common/git_util.py", line 230, in deleted_files
    in self.repo.remote(name=remote).refs[branch].commit.diff(
  File "/Users/morita/.pyenv/versions/3.9.0/lib/python3.9/site-packages/git/util.py", line 891, in __getitem__
    raise IndexError("No item found with id %r" % (self._prefix + index)) from e
IndexError: No item found with id 'origin/master'
MoritanoMacBook-Pro:Demisto_git morita$ `

3. please advice how to do the unit test. This is the first time for me to be told to do so..

morita39 commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.py:
```diff
@@ -1674,8 +1685,8 @@ def build_member_data(result: dict, readable_output: str, printable_result: dict
                                                  members_printable_result,
-                                                 ['member-name', 'member-uid', 'member-type',
-                                                  'member-domain-name', 'member-domain-uid'],
+                                                 ['member-name', 'member-uid', 'member-type''member-ipv4-address',
```
Added ','

morita39 reviewed (commented):
updated

morita39 commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.py:
```diff
@@ -366,8 +366,6 @@ def checkpoint_add_host_command(client: Client, name, ip_address, ignore_warning
     name = argToList(name)
     ip_address = argToList(ip_address)
     groups = argToList(groups)
-    ignore_warnings = argToBool(ignore_warnings)
```
argToBool is not defined error occurred. Please advise how to deal with this.

ShahafBenYakir replied:
If you are getting this error while working on your XSOAR instance is because you have an outdated `Base` pack, let's update it. If you get it while running the script locally it because your CommonServerPython is not up to date.

morita39 replied:
Base pack is now 1.10.27 358734 with warnings "Pack with ID 'DrawRelatedIncidentsCanvas' and name 'DrawRelatedIncidentsCanvas' is not comatpbile with environment version and will be ignored", but still "Failed to execute checkpoint-host-delete command. Error: name 'argToBool' is not defined" error persists. How do I update CommonServerPython?

morita39 commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.py:
```diff
@@ -366,8 +366,6 @@ def checkpoint_add_host_command(client: Client, name, ip_address, ignore_warning
     name = argToList(name)
     ip_address = argToList(ip_address)
     groups = argToList(groups)
```
argToBool is not defined error occurred. Please advise how to deal with this.

morita39 reviewed (commented):
update

morita39 commented on Packs/CheckpointFirewall/ReleaseNotes/2_0_8.md:
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### CheckPointFirewall_v2
+- Added *ignore_warnings* and *ignore_errors* arguments to **checkpoint-host-add**.
```
Added *ignore_warnings* and *ignore_errors* arguments to **checkpoint-host-delete** and **checkpoint-host-add**

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -129,36 +109,38 @@ script:
+      predefined:
+      - "true"
+      - "false"
+      description: Whether to ignore warnings when adding host
```
```suggestion
      description: Whether to ignore warnings when adding a host.
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -129,36 +109,38 @@ script:
+      predefined:
+      - "true"
+      - "false"
+      description: Whether to ignore error when adding host
```
```suggestion
      description: Whether to ignore errors when adding a host.
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -300,54 +250,47 @@ script:
+      predefined:
+      - "true"
+      - "false"
+      description: Whether to ignore warnings when delete host
```
```suggestion
      description: Whether to ignore warnings when deleting a host.
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -300,54 +250,47 @@ script:
+      predefined:
+      - "true"
+      - "false"
+      description: Whether to ignore errors when delete host
```
```suggestion
      description: Whether to ignore errors when deleting a host.
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -1238,92 +938,59 @@ script:
+      defaultValue: None
+    - name: new_position
       description: New position in the rulebase. Value can be "int" to set a specific
         position, or str- 'top' or 'bottom'
```
```suggestion
        position, or str- 'top' or 'bottom'.
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/README.md:
```diff
@@ -228,8 +234,8 @@ Add new host
 | session_id | Execute command with a specific session ID | Required | 
-
-
+| ignore_warnings | Whether to ignore warnings when adding host | Optional |
```
```suggestion
| ignore_warnings | Whether to ignore warnings when adding a host. | Optional |
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/README.md:
```diff
@@ -228,8 +234,8 @@ Add new host
-
-
+| ignore_warnings | Whether to ignore warnings when adding host | Optional |
+| ignore_errors | Whether to ignore error when adding host | Optional |
```
```suggestion
| ignore_errors | Whether to ignore errors when adding a host. | Optional |
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/README.md:
```diff
@@ -301,7 +307,8 @@ update host changes
 | groups | Collection of group identifiers. | Optional | 
 | session_id | Execute command with a specific session ID | Required | 
-
+| ignore_warnings | Whether to ignore warnings when adding host | Optional |
```
```suggestion
| ignore_warnings | Whether to ignore warnings when adding a host. | Optional |
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/README.md:
```diff
@@ -301,7 +307,8 @@ update host changes
 | session_id | Execute command with a specific session ID | Required | 
-
+| ignore_warnings | Whether to ignore warnings when adding host | Optional |
+| ignore_errors | Whether to ignore error when adding host | Optional |
```
```suggestion
| ignore_errors | Whether to ignore errors when adding a host. | Optional |
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/README.md:
```diff
@@ -537,7 +544,8 @@ Get all data of a given group
 | CheckPoint.Group.last-modifier | String | indicates the last user modified the object | 
 | CheckPoint.Group.read-only | Boolean | indicates if the object is read only | 
-
+| CheckPoint.Group.members.ipv4-address | String | ipv4-address |
```
```suggestion
| CheckPoint.Group.members.ipv4-address | String | Group members IPv4 addresses |
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/README.md:
```diff
@@ -537,7 +544,8 @@ Get all data of a given group
 | CheckPoint.Group.read-only | Boolean | indicates if the object is read only | 
-
+| CheckPoint.Group.members.ipv4-address | String | ipv4-address |
+| CheckPoint.Group.members.ipv6-address | String | ipv6-address |
```
```suggestion
| CheckPoint.Group.members.ipv6-address | String | Group members IPv6 addresses |
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -404,24 +338,23 @@ script:
-  - arguments:
-    - default: false
+    - contextPath: CheckPoint.Group.members.ipv4-address
+      description: ipv4-address
```
```suggestion
      description: Group members IPv4 addresses.
```

roysagi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -404,24 +338,23 @@ script:
+      description: ipv4-address
+      type: string
+    - contextPath: CheckPoint.Group.members.ipv6-address
+      description: ipv6-address
```
```suggestion
      description: Group members IPv6 addresses.
```

yaakovi commented on Packs/CheckpointFirewall/Integrations/CheckPointFirewall_v2/CheckPointFirewall_v2.yml:
```diff
@@ -1,66 +1,52 @@
 - display: Username
-  name: username
-  required: true
+  name: userame
```
```suggestion
  name: username
```

Noy-Maimon commented:
@cincrement Hi, can you please sign the CLA form here? since you have commits related to this issue, it is needed you sign this form: https://cla-assistant.io/demisto/content?pullRequest=12724

Noy-Maimon commented:
Due to issues with git, reopened in this PR: https://github.com/demisto/content/pull/12825
